### PR TITLE
Maemo/stretch

### PR DESCRIPTION
**Pull request by spinal84**
* Add systemd support
* Add cross-compiling support
* Fix warnings
* Minor fixes

### CODE REVIEW

File: modules/runlevel.c [hunk 1]
```diff
@@ -81,8 +81,30 @@ static void shutdown(dsme_runlevel_t runlevel)
            runlevel == DSME_RUNLEVEL_REBOOT   ? "Reboot"   :
                                                 "Malf");
 
+  /* If we have systemd, use systemctl commands */
```
Comment (MerlijnWajer) [inline modules/runlevel.c:84]: Should this also go in `#ifdef DSME_SYSTEMD_ENABLE` ?

File: debian/control [hunk 2]
```diff
@@ -11,6 +11,7 @@ Build-Depends: debhelper (>= 10),
  libdbus-1-dev (>= 1.0),
  libdbus-glib-1-dev,
  maemo-system-services-dev,
+ libsystemd-dev,
```
Comment (MerlijnWajer) [inline debian/control:14]: Is there a way to make this optional (will this somehow cause trouble in Devuan)?

Comment (spinal84) [reply debian/control:14]: It doesn't add troubles to Devuan. I already tested this to work in ascii-devel.
Devuan has support for libsystemd-dev and many packages in Devuan are linked to libsystemd0.
The purpose of Devuan is to not use systemd as init system, but it still uses it's libraries.
Anyway, this is just a build-time dependency which doesn't affect the resulting package for Devuan in any way.

Comment (MerlijnWajer) [reply debian/control:14]: Thanks for clarifying.

File: modules/runlevel.c (hunk 1 above)
Comment (spinal84) [reply modules/runlevel.c:84]: This was done for simplicity. And I tried to stick the decisions made by Mer developers to make our code closer to what they have.

If you don't like this part, I can rewrite it.

Comment (MerlijnWajer) [reply modules/runlevel.c:84]: No, it's good to keep it close to what Mer project does. Maybe one day we can just use their mer version.

**Comment by MerlijnWajer** [comment]
Looks good to me. @freemangordon - got any opinions?

File: debian/control (hunk 2 above)
Comment (parazyd) [reply debian/control:14]: FWIW, in Devuan we also have libraries that can replace libsystemd and keep API compatibility.

**Pull request merged**